job_template.go

```diff
@@ -52,18 +54,42 @@ func (jt *JobTemplateService) Launch(id int, data map[string]interface{}, params
 		return nil, err
 	}
 
-	// in case invalid job id return
-	if result.Job == 0 {
-		return nil, errors.New("invalid job id 0")
+	return result, nil
+}
+
+// CreateJobTemplateCallBack executes a PATCH HTTP Request to create the callback url and the generated host_config_key
+func (jt *JobTemplateService) CreateJobTemplateCallBack(template *JobTemplate) (*JobTemplate, error) {
+	if template.ID == 0 {
+		return nil, fmt.Errorf("Job template ID must be passed")
 	}
 
-	return result, nil
+	endpoint := "/api/v2/job_templates/" + strconv.Itoa(template.ID)
+	template.AllowCallbacks = true
+	template.HostConfigKey = uuid.NewV4().String()
```

Comment:
Why we need to set these two by manually?

Reply:
I do not know, the original commit : https://github.com/davidfischer-ch/awx-go/commit/72fdcdc204d8bbf328cff0ab43e35b7aab727c6c.

Reply:
CreateJobTemplateCallBack executes a PATCH HTTP Request to create the callback url and the generated host_config_key.

The values are set to get that result. OK?